docs/Observability/Observability.md

```diff
@@ -0,0 +1,161 @@
+
+  ```yaml
+  <component>:
+    metrics:
+      enabled: true 
+      serviceMonitor:
+        enabled: true # when using the Prometheus Operator
+  ```
+The default port for collecting metrics in AI DIAL components is 9464. You can change the parameter `<component>.containerPorts.metrics` to change the default port.
+
+## OpenTelemetry
+
+[OpenTelemetry](https://opentelemetry.io/) is an open-source observability framework designed to standardize the collection of telemetry data across distributed systems. By providing a unified set of APIs, libraries, and agents, it enables developers to capture traces, metrics, and logs from their applications seamlessly. OpenTelemetry simplifies the monitoring process and enhances visibility into application performance and reliability, making it easier to troubleshoot issues and optimize systems in real-time.
+
+AI DIAL supports OpenTelemetry (OTEL) methods to enhance observability by providing powerful metrics for collection and tracing capabilities, enabling deeper insights into system performance and behavior.
+
+### AI DIAL Configuration
+
+All environment variables you can find in the official OpenTelemetry [documentation](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/).
+
+#### Python Components
+
+All standard python environment variables you can find in the official OpenTelemetry [documentation](https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/logging/logging.html). 
+
+* If the value for **OTEL_METRICS_EXPORTER** is not set, the [OpenTelemetry Prometheus Metric Exporter](https://www.npmjs.com/package/@opentelemetry/exporter-prometheus) will be used. 
+* If its value is set to `"otlp"`, the [OpenTelemetry Collector Metrics Exporter for Web and Node](https://www.npmjs.com/package/@opentelemetry/exporter-metrics-otlp-http) will be used.
+
+Example configuration of OpenTelemetry:
+
+```yaml
+  OTEL_RESOURCE_ATTRIBUTES: "service.name=<service_name>" # Key-value pairs to be used as resource attributes
+  OTEL_EXPORTER_OTLP_ENDPOINT: "<otlp_endpoint_url>" # OTEL endpoint URL
+  OTEL_LOGS_EXPORTER: "otlp" # logs exporter to be used
+  OTEL_METRICS_EXPORTER: "otlp|otlp,prometheus" # metrics exporter to be used
+  OTEL_TRACES_EXPORTER: "otlp" # trace exporter to be used
+  OTEL_PYTHON_LOG_CORRELATION: "true|false" # enable trace context injection
+  OTEL_PYTHON_FASTAPI_EXCLUDED_URLS: "<exclude_url>" # to exclude certain URLs from tracking
+```
+
+#### Node.js Components
+
+* If the value for **OTEL_METRICS_EXPORTER** is not set, the [OpenTelemetry Prometheus Metric Exporter](https://www.npmjs.com/package/@opentelemetry/exporter-prometheus) will be used. 
+* If its value is set to `"otlp"`, the [OpenTelemetry Collector Metrics Exporter for Web and Node](https://www.npmjs.com/package/@opentelemetry/exporter-metrics-otlp-http) will be used.
+
+Example configuration of OpenTelemetry:
+
```

Comment:
This is configuration for what? Core?

Reply:
All components that use the python ai-dial-sdk library.